packages/amplify-e2e-core/src/utils/nexpect.ts

```diff
@@ -160,7 +160,7 @@ function chain(context: Context): ExecutionContext {
 
     wait(
       expectation: string | RegExp,
-      callback = (data: string) => {
+      callback: (data: string) => void = () => {
```

Comment:
Does this function parameter not need a default value?

Reply:
this thing here is a default no-op implementation of https://github.com/aws-amplify/amplify-cli/blob/bf63580826133c56c56f86a59ecb918edfde8002/packages/amplify-e2e-core/src/utils/nexpect.ts#L75C20-L75C20

The code is correct. Parameter is ignored in both old and new implementation. But removing explicit parameter require typing `callback` parameter otherwise it's not matching `ExecutionContext` type.

Reply:
This is legal in typescript.
```
let foo: (param: string) => void;
foo = () => {};
```

I.e. function must accept params from typing or less (less means it ignores inputs).